### PR TITLE
Update all HTML for indexterm, and create ix.html. 

### PR DESCRIPTION
**Pull request by gitvsh**
- Several upgrades and bug fixes.
- Automatic edit of all indexterm references.
- Generate HTML for ix.html.

This change has two separate parts (but it makes no difference that they are together):

1. Edit the existing HTML files to give id= values to all indexterm entries that don't have them. This is necessary to be able to reference from an index entry to all separate terms.
    - Some indexterm entries had no id but a startref value which associated with another indexterm which was the start point for the indexterm without an id (a bit of a backwards naming, but not important). I've made up new id values in a way that allows to associate the pair of references, as a `start-end` representation (below).
    - Other indexterm entries had no id and nothing else, but because they had the same values for primary and secondary labels, it was clear they were just additional points that term was found. I've made up new id values for them, but without overlap with the above case; these appear as "start" in the index itself.
2. Generating the ix.html from the edited existing HTML whose _all_ indexterm entries now have an id.
    - The index is broken up into sections by the alphabetic letters. I used "Letter H" instead of just "H" to allow the reader to search for a section start by typing "letter h" rather than "h" which matches for all "h" in the index.
    - An index entry starts with a non-linkified term text, indented for secondary terms, then followed by one or more `start` or `start-end`, all separated by commas. The effect is similar to a printed book, but instead of page numbers (which HTML format doesn't have) we use "start" for a single page or `start-end` for a range of pages (e.g. 170-172).

We can later revise how we represent the pages, e.g. to make it more clear which chapter the reference is to, but this is a decent enough start. :)

### CODE REVIEW

File: tools/index-updater.awk [hunk 1]
```diff
@@ -171,25 +197,31 @@ END {
...
-	print "</li>";
 
 	primary = data_pri;
+	secondary = data_sec;
+	entries += 1;
     }
+    print "</li>";  # The lazy assumption that the index is not empty.
 
+    print ""
+    print "<!-- Index generated by tools/index-updater.awk, has " entries " entries. -->";
```
Comment (lasagnaboy) [inline tools/index-updater.awk:215]: nice

File: tools/index-updater.awk [hunk 2]
```diff
@@ -133,7 +155,7 @@ END {
 		    level = level - 1;
 		}
 		print "";
-		print "  <h3>" letter_new "</h3>";
+		print "  <h3> Letter " letter_new "</h3>";
```
Comment (lasagnaboy) [inline tools/index-updater.awk:158]: I don't find that "Letter" adds to the readability of the index; I rather prefer
`A`
to
`Letter A`

**Review by lasagnaboy — APPROVED** [review]
Read through some of the generated HTML;
one comment about how index-updater.awk formats the index headers.

File: tools/index-updater.awk (hunk 2 above)
Comment (gitvsh) [reply tools/index-updater.awk:158]: The reason for "Letter " is covered in the PR description, but I also don't love what I did. I have an idea how to solve the underlying issue, and will do so in a follow change.

**Comment by gitvsh** [comment]
Thank you for the review. I'll address your one suggestion in a followup, as explained.

**Pull request merged**